### PR TITLE
Bye bye nova

### PR DESCRIPTION
**Pull request by hmlanigan**
## Checklist

 - [x] Checked if it requires a [pylibjuju](https://github.com/juju/python-libjuju) change?
 - [notneeded] Added [integration tests](https://github.com/juju/juju/tree/develop/tests) for the PR?
 - [nonavailable] Added or updated [doc.go](https://discourse.jujucharms.com/t/readme-in-packages/451) related to packages changed?
 - [x] Do comments answer the question of why design decisions were made?

----

## Description of change

The Nova Networking component of OpenStack has been deprecated since the Newton release and will be removed in the Ussuri release in May 2020.  Its replacement, Neutron has been supported in juju for a few years.  Remove 

More work can be done to remove support from the goose library.

## QA steps

Bootstrap an OpenStack config.  If available, without Neutron networking to verify the error message.

## Documentation changes

Add to release notes for 2.8

### CODE REVIEW

File: provider/openstack/provider.go [hunk 1]
```diff
@@ -703,17 +703,15 @@ func (e *Environ) PrepareForBootstrap(ctx environs.BootstrapContext, controllerN
 		return err
 	}
 	if !e.supportsNeutron() {
-		logger.Warningf(`Using deprecated OpenStack APIs.
+		logger.Warningf(`Using unsupported OpenStack APIs.
```
Comment (hmlanigan) [inline provider/openstack/provider.go]: @mitechie: Thoughts on wording?

**Comment by hmlanigan** [comment]
!!build!!

**Comment by hmlanigan** [comment]
Removing switchingNetwork{} is trivial.  Configuring our tests correctly to run while verifying that neutron is supported is not.  Either the tests have to be hacked up to behave like the Cinder ones, or the goose identity test service requires updates to return the endpoint map.

File: provider/openstack/provider.go (hunk 1 above)
Comment (mitechie) [reply provider/openstack/provider.go]: Let's go ahead and make this an error since it's a bad bad thing that we know won't work.

**Comment by hmlanigan** [comment]
!!build!!

**Comment by hmlanigan** [comment]
$$merge$$

**Pull request merged**